pkg/build/admission/overrides/admission.go

```diff
@@ -0,0 +1,84 @@
+package overrides
+
+import (
+	"io"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/admission"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+
+	buildadmission "github.com/openshift/origin/pkg/build/admission"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+)
+
+func init() {
+	admission.RegisterPlugin("BuildOverrides", func(c kclient.Interface, config io.Reader) (admission.Interface, error) {
+		overridesConfig, err := getConfig(config)
+		if err != nil {
+			return nil, err
+		}
+
+		glog.V(4).Infof("Initializing BuildOverrides plugin with config: %#v", overridesConfig)
+		return NewBuildOverrides(overridesConfig), nil
+	})
+}
+
+func getConfig(in io.Reader) (*BuildOverridesConfig, error) {
+	overridesConfig := &BuildOverridesConfig{}
+	err := buildadmission.ReadPluginConfig(in, overridesConfig)
+	if err != nil {
+		return nil, err
+	}
+	return overridesConfig, nil
+}
+
+type buildOverrides struct {
+	*admission.Handler
+	overridesConfig *BuildOverridesConfig
+}
+
+// NewBuildOverrides returns an admission control for builds that overrides
+// settings on builds
+func NewBuildOverrides(overridesConfig *BuildOverridesConfig) admission.Interface {
+	return &buildOverrides{
+		Handler:         admission.NewHandler(admission.Create, admission.Update),
+		overridesConfig: overridesConfig,
+	}
+}
+
+// Admit appplies configured overrides to a build in a build pod
+func (a *buildOverrides) Admit(attributes admission.Attributes) error {
+	if a.overridesConfig == nil {
+		return nil
+	}
+	if !buildadmission.IsBuildPod(attributes) {
+		return nil
+	}
+	build, version, err := buildadmission.GetBuild(attributes)
+	if err != nil {
+		return err
+	}
+
+	glog.V(4).Infof("Handling build %s/%s", build.Namespace, build.Name)
+
+	a.applyOverrides(build)
+
+	return buildadmission.SetBuild(attributes, build, version)
+}
+
+func (a *buildOverrides) applyOverrides(build *buildapi.Build) {
+	if a.overridesConfig.ForcePull {
+		if build.Spec.Strategy.DockerStrategy != nil {
+			glog.V(5).Infof("Setting docker strategy ForcePull to true in build %s/%s", build.Namespace, build.Name)
+			build.Spec.Strategy.DockerStrategy.ForcePull = true
+		}
+		if build.Spec.Strategy.SourceStrategy != nil {
+			glog.V(5).Infof("Setting source strategy ForcePull to true in build %s/%s", build.Namespace, build.Name)
+			build.Spec.Strategy.SourceStrategy.ForcePull = true
+		}
+		if build.Spec.Strategy.CustomStrategy != nil {
+			glog.V(5).Infof("Setting custom strategy ForcePull to true in build %s/%s", build.Namespace, build.Name)
+			build.Spec.Strategy.CustomStrategy.ForcePull = true
```

Comment:
this isn't sufficient to get forcepull behavior on a custom build.  you need to actually change the pod itself to be a pullAlways strategy:

https://github.com/openshift/origin/blob/master/pkg/build/controller/strategy/custom.go#L81-L86

(this is because custom build is a special case where we're not going to run something else on the user's behalf, we're going to run *this* on the user's behalf)